tests/phpunit/tests/admin/wpListTable.php

```diff
@@ -0,0 +1,226 @@
+
+		parent::set_up_before_class();
+
+		require_once ABSPATH . 'wp-admin/includes/class-wp-list-table.php';
+
+		$hook_suffix      = '_wp_tests';
+		self::$list_table = new WP_List_Table();
+	}
+
+	/**
+	 * Tests the "get_views_links()" method.
+	 *
+	 * @ticket 42066
+	 *
+	 * @covers WP_List_Table::get_views_links
+	 *
+	 * @dataProvider data_get_views_links
+	 *
+	 * @param array $link_data {
+	 *     An array of link data.
+	 *
+	 *     @type string $url     The link URL.
+	 *     @type string $label   The link label.
+	 *     @type bool   $current Optional. Whether this is the currently selected view.
+	 * }
+	 * @param array $expected
+	 */
+	public function test_get_views_links( $link_data, $expected ) {
+		$get_views_links = new ReflectionMethod( self::$list_table, 'get_views_links' );
+		$get_views_links->setAccessible( true );
+
+		$actual = $get_views_links->invokeArgs( self::$list_table, array( $link_data ) );
+
+		$this->assertSameSetsWithIndex( $expected, $actual );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_get_views_links() {
+		return array(
+			'one "current" link'  => array(
+				'link_data' => array(
+					'all'       => array(
+						'url'     => 'https://example.org/',
+						'label'   => 'All',
+						'current' => true,
+					),
+					'activated' => array(
+						'url'     => add_query_arg( 'status', 'activated', 'https://example.org/' ),
+						'label'   => 'Activated',
+						'current' => false,
+					),
+				),
+				'expected'  => array(
+					'all'       => '<a href="https://example.org/" class="current" aria-current="page">All</a>',
+					'activated' => '<a href="https://example.org/?status=activated">Activated</a>',
+				),
+			),
+			'two "current" links' => array(
+				'link_data' => array(
+					'all'       => array(
+						'url'     => 'https://example.org/',
+						'label'   => 'All',
+						'current' => true,
+					),
+					'activated' => array(
+						'url'     => add_query_arg( 'status', 'activated', 'https://example.org/' ),
+						'label'   => 'Activated',
+						'current' => true,
+					),
+				),
+				'expected'  => array(
+					'all'       => '<a href="https://example.org/" class="current" aria-current="page">All</a>',
+					'activated' => '<a href="https://example.org/?status=activated" class="current" aria-current="page">Activated</a>',
+				),
+			),
+			'one "current" link and one without "current" key' => array(
+				'link_data' => array(
+					'all'       => array(
+						'url'     => 'https://example.org/',
+						'label'   => 'All',
+						'current' => true,
+					),
+					'activated' => array(
+						'url'   => add_query_arg( 'status', 'activated', 'https://example.org/' ),
+						'label' => 'Activated',
+					),
+				),
+				'expected'  => array(
+					'all'       => '<a href="https://example.org/" class="current" aria-current="page">All</a>',
+					'activated' => '<a href="https://example.org/?status=activated">Activated</a>',
+				),
```

Comment:
```suggestion
					'activated' => array(
						'url'   => add_query_arg( 'status', 'activated', 'https://example.org/' ),
						'label' => 'Activated',
					),
					'my activated' => array(
						'url'   => add_query_arg( array( 'author' => 1, 'status' => 'activated' ), 'https://example.org/' ),
						'label' => 'My activated',
					),
				),
				'expected'  => array(
					'all'       => '<a href="https://example.org/" class="current" aria-current="page">All</a>',
					'activated' => '<a href="https://example.org/?status=activated">Activated</a>',
					'my activated' => '<a href="https://example.org/?status=activated&amp;author=1">Activated</a>',
				),
```

Something to ensure it's escaped for output -- I think `esc_url()` escapes to `&amp;` but it would be good to check.